fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/impl/resource/conditions/OverlayConditionsMetadata.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.resource.conditions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import net.minecraft.resource.metadata.ResourceMetadataSerializer;
+import net.minecraft.util.JsonHelper;
+
+import net.fabricmc.fabric.api.resource.conditions.v1.ResourceConditions;
+
+public record OverlayConditionsMetadata(JsonArray overlays) {
+	private static final Pattern DIRECTORY_NAME_PATTERN = Pattern.compile("[-_a-zA-Z0-9.]+");
+	public static final ResourceMetadataSerializer<OverlayConditionsMetadata> SERIALIZER = new ResourceMetadataSerializer<>() {
```

Comment:
I don't agree with the fact that it's in the impl package because it makes working with the datagen difficult. Look at `MetadataProvider`

Reply:
This also means that you should replace `JsonArray` with a separate class, similar to what vanilla does in `PackOverlaysMetadata`